spec/02-integration/05-proxy/01-router_spec.lua

```diff
@@ -499,4 +499,44 @@ describe("Router", function()
         check(i, args[3], args[4]))
     end
   end)
+
+  describe("url encoded uris", function()
+
+    setup(function()
+      helpers.dao:truncate_tables()
+      assert(helpers.dao.apis:insert {
+        name         = "localbin",
+        upstream_url = "http://localhost:9999/",
+        uris         = {
+          "/",
+        },
+        hosts        = {
+          "localbin.com",
+        },
+      })
+
+      assert(helpers.start_kong {
+        nginx_conf = "spec/fixtures/custom_nginx.template",
```

Comment:
> It would take me about a day to move all our httpbin.org tests to that localbin

Yes, doing so is quite easy, but let's move there once it's properly done, and not before, or else we'll be adding overhead when doing so, or potentially lack discipline and end up lagging behind in implementing it, and instead keep using that fixture template and lose atomicity in our tests. It just makes things entangled and messy. Such a localbin interface also has requirements. For example we need something like:
```lua
kong.start {
  http_config = [[
    location /custom {

    }
  ]]
}
```

for occasions where we need an endpoint only once. This is the idea behind the short-term "localbin". There are other requirements.

FYI, most of the httpbin we use is already there as an ngx_lua implementation: https://github.com/thibaultcha/test-nginx-httpbin/blob/master/lib/Test/Nginx/Socket/Httpbin.pm#L66-L232